From 8845ad6cac28d773b295e5775c2284a08e17cf47 Mon Sep 17 00:00:00 2001
From: Sebastien Guiriec <s-guiriec@ti.com>
Date: Tue, 5 Apr 2011 14:38:53 +0200
Subject: [PATCH 10/12] ASoC: ABE HAL: Correct port status update for dmareq port

ABE port status was initialize to early. This was introducing a
missing initialization of ASRC for Voice port in case of MPU
recording/playing (not for MODEM)

Signed-off-by: Sebastien Guiriec <s-guiriec@ti.com>
---
 sound/soc/omap/abe/abe_api.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

Index: kernel/sound/soc/omap/abe/abe_api.c
===================================================================
--- kernel.orig/sound/soc/omap/abe/abe_api.c	2011-04-07 00:20:27.147134606 +0900
+++ kernel/sound/soc/omap/abe/abe_api.c	2011-04-07 15:14:22.388330637 +0900
@@ -946,12 +946,14 @@
 	abe_port[id].protocol.p.prot_dmareq.iter = abe_dma_port_iteration(f);
 	abe_port[id].protocol.p.prot_dmareq.dma_addr = ABE_DMASTATUS_RAW;
 	abe_port[id].protocol.p.prot_dmareq.dma_data = (1 << d);
-	abe_port[id].status = OMAP_ABE_PORT_INITIALIZED;
 	/* load the dma_t with physical information from AE memory mapping */
 	abe_init_dma_t(id, &((abe_port[id]).protocol));
+
 	/* load the micro-task parameters */
 	abe_init_io_tasks(id, &((abe_port[id]).format),
 			  &((abe_port[id]).protocol));
+	abe_port[id].status = OMAP_ABE_PORT_INITIALIZED;
+
 	/* load the ATC descriptors - disabled */
 	abe_init_atc(id);
 	/* return the dma pointer address */
@@ -1092,6 +1094,7 @@
 	abe_read_hardware_configuration(UC_NULL, &OPP, &CONFIG);
 	(abe_port[id]).protocol.p.prot_serial.iter =
 		abe_dma_port_iter_factor(f);
+
 	/* load the micro-task parameters */
 	abe_init_io_tasks(id, &((abe_port[id]).format),
 			  &((abe_port[id]).protocol));
